libraries/aws_flow_log.rb

```diff
@@ -0,0 +1,85 @@
+class AwsFlowLog < Inspec.resource(1)
+  name 'aws_flow_log'
+  desc 'Verifies settings for an IAM Flow Log'
+  example "
+    describe aws_flow_log(flow_log_id: 'fl-12345678') do
+      its('flow_log_status') { should eq 'ACTIVE' }
+      its('traffic_type') { should eq 'ALL' }
+      its('deliver_logs_error_message') { should eq 'Access error' }
+    end
+  "
+
+  include AwsResourceMixin
+  attr_reader :flow_log_id, :vpc_id, :subnet_id, :resource_id, :flow_log_status,
+              :deliver_logs_error_message, :deliver_logs_permission_arn,
+              :deliver_logs_status, :log_group_name, :traffic_type
+
+  def to_s
+    "Flow Log #{@flow_log_id}"
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [:flow_log_id, :vpc_id, :subnet_id],
+      allowed_scalar_name: :flow_log_id,
+      allowed_scalar_type: String,
+    )
+    if validated_params.empty?
+      raise ArgumentError, 'You must provide a flow_log_id to aws_flow_log.'
+    end
+    validated_params
+  end
+
+  def fetch_from_aws
+    @resource_id = vpc_id ? vpc_id : subnet_id
+    filters_array = []
```

Comment:
This block can be done much more smoothly:

```
filters = []
filters.push { name: 'flow-log-id', values: [ @flow_log_id ] } unless @flow_log_id.nil?
filters.push { name: 'resoource-id', values: [ @resource_id ] } unless @resource_id.nil?
```
And then you're ready for the describe_flow_logs call.